src/smart-components/group/groups.js

```diff
@@ -176,10 +179,25 @@ const Groups = () => {
         ]
       : []),
   ];
+
   const data = groups.map((group) =>
     group.platform_default || group.admin_default ? { ...group, principalCount: `All${group.admin_default ? ' org admins' : ''}` } : group
   );
-  const rows = createRows(isAdmin, data, selectedRows);
+
+  const fetchExpandedRoles = (uuid) => dispatch(fetchRolesForGroup(uuid, defaultCompactSettings));
+  const fetchExpandedMembers = (uuid) => dispatch(fetchMembersForGroup(uuid, defaultCompactSettings));
+
+  const onExpand = (_event, _rowIndex, colIndex, isOpen, rowData) => {
+    if (!isOpen) {
+      setExpanded({ ...expanded, [rowData.uuid]: colIndex + Number(!isAdmin) });
+      colIndex + Number(!isAdmin) === 2 && fetchExpandedRoles(rowData.uuid);
+      colIndex + Number(!isAdmin) === 3 && fetchExpandedMembers(rowData.uuid);
+    } else {
+      setExpanded({ ...expanded, [rowData.uuid]: -1 });
+    }
+  };
+
+  const rows = createRows(isAdmin, data, selectedRows, expanded, isLoading);
```

Comment:
this is probably not the isLoading flag we want - it's for `groups` data but we need to store another flag for group roles and members. Then it would be part of data (something like `group.roles.isLoading`) so no need to pass it separately